relay-server/src/actors/processor/span/processing.rs

```diff
@@ -150,6 +142,43 @@ pub fn extract_from_event(state: &mut ProcessEnvelopeState) {
         state.managed_envelope.envelope_mut().add_item(item);
     };
 
+    // Check feature flag.
+    if !state
+        .project_state
+        .has_feature(Feature::SpanMetricsExtraction)
+    {
+        // When span metrics extraction is disabled, we need to check if DDM is enabled
+        // to still index spans with _metrics_summary
+        if !state.project_state.has_feature(Feature::CustomMetrics) {
+            return;
+        }
+
+        let Some(event) = state.event.value() else {
+            return;
+        };
+
+        // Metrics summary is empty so we don't need to ingest the span.
+        if event._metrics_summary.is_empty() {
+            return;
+        }
+
+        // Extract transaction as a span.
+        let mut transaction_span: Span = event.into();
+
+        // Extract tags to add to this span as well
+        let shared_tags = tag_extraction::extract_shared_tags(event);
+        transaction_span.sentry_tags = Annotated::new(
+            shared_tags
+                .clone()
+                .into_iter()
+                .map(|(k, v)| (k.sentry_tag_key().to_owned(), Annotated::new(v)))
+                .collect(),
+        );
+        add_span(transaction_span.into());
```

Comment:
Even if we need this extra logic (see comment below), I think we could prevent copying this snippet and instead have something like

```rust
let extract_transaction_span = extraction_enabled || (custom_metrics_enabled && !summary.is_empty());
let extract_child_spans = extraction_enabled;
```

Reply:
True. I refactored it.